### PR TITLE
pair + nearMap

### PR DESCRIPTION
**Pull request by JavaPsyh**

### CODE REVIEW

File: src/tasks/like/map/Storage.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            }
+        }
+        return result;
+    }
+
+    public class Something<T, V> {
+        private T key;
+        private V something;
+
+        public Something(T i, V something) {
```
Comment (Sofasmile) [inline src/tasks/like/map/Storage.java]: better change:
i -> key
something -> value

File: src/tasks/like/map/Storage.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        V result = null;
+        for (int i = 0; i < indexForNextPut; i++) {
+            if (key == values[i].getKey()) {
+                result = values[i].getSomething();
+            }
+        }
+        return result;
+    }
+
+    public class Something<T, V> {
```
Comment (Sofasmile) [inline src/tasks/like/map/Storage.java]: move this class into separate file

File: src/tasks/like/map/Storage.java (hunk 1 above)
Comment (Sofasmile) [reply src/tasks/like/map/Storage.java]: better create other name for class

File: src/tasks/like/map/Storage.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package tasks.like.map;
+
+public class Storage<T, V> {
+
+    private Something<T, V>[] values = new Storage.Something[5];
```
Comment (Sofasmile) [inline src/tasks/like/map/Storage.java]: magic number

File: src/tasks/like/map/Storage.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    public int getIndexForNextPut() {
+        return indexForNextPut;
+    }
+
+    public void put(T key, V value) {
+
+        if (indexForNextPut == values.length) {
+            Something[] temp;
+            temp = new Something[values.length + (values.length / 2)];
+            for (int i = 0; i < values.length; i++) {
```
Comment (Sofasmile) [inline src/tasks/like/map/Storage.java]: you can use System.arraycopy()

File: src/tasks/stepic/pair/Pair.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        if (o == null || getClass() != o.getClass()) return false;
+        Pair<?, ?> pair = (Pair<?, ?>) o;
+        return Objects.equals(first, pair.first) &&
+                Objects.equals(second, pair.second);
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        final int theAnswerToTheUltimateQuestionOfLifeTheUniverseAndEverything = 42;
```
Comment (Sofasmile) [inline src/tasks/stepic/pair/Pair.java]: rename variable

File: src/tasks/like/map/StorageCreatorApp.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        System.out.println(storage1.get(50));
+        System.out.println();
+        Storage<String, Integer> storage2 = new Storage<>();
+        storage2.put("Katia", 35);
+        storage2.put("Zina", 32);
+        storage2.put("Vera", 37);
+        storage2.put("Sofia", 18);
+        System.out.println(storage2.get("Katia"));
+        System.out.println(storage2.get("Sofia"));
+
```
Comment (Sofasmile) [inline src/tasks/like/map/StorageCreatorApp.java]: cs

**Review by Sofasmile — COMMENTED** [review]
resolve comments

File: src/tasks/like/map/Storage.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
+package tasks.like.map;
+
+public class Storage<T, V> {
```
Comment (Sofasmile) [inline src/tasks/like/map/Storage.java:3]: Ctrl+Alt+L

File: src/tasks/like/map/Storage.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+    private final int DEFAULT_CAPACITY = 5;
```
Comment (Sofasmile) [inline src/tasks/like/map/Storage.java]: add static to make this field constant

File: src/tasks/like/map/Storage.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+    public void arrayGrower() {
+        Depository[] temp;
+        temp = new Depository[values.length + (values.length >> 1)];
+        System.arraycopy(values, 0, temp, 0, values.length);
+        values = temp;
+    }
+
+    public void put(T key, V value) {
+
```
Comment (Sofasmile) [inline src/tasks/like/map/Storage.java]: cs

File: src/tasks/stepic/pair/Pair.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    public String toString() {
+        return "Pair{" +
+                "first=" + first +
+                ", second=" + second +
+                '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```
Comment (Sofasmile) [inline src/tasks/stepic/pair/Pair.java]: better use your implementation

File: src/tasks/stepic/pair/Pair.java [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+        if (o == null || getClass() != o.getClass()) return false;
+        Pair<?, ?> pair = (Pair<?, ?>) o;
+        return Objects.equals(first, pair.first) &&
+                Objects.equals(second, pair.second);
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        final int theFortyTwoAnswer = 42;
```
Comment (Sofasmile) [inline src/tasks/stepic/pair/Pair.java:58]: it is not the best name of the variable
but, it better than previous

**Review by Sofasmile — COMMENTED** [review]
resolve comments

File: src/tasks/like/map/Storage.java (hunk 7 above)
Comment (JavaPsyh) [reply src/tasks/like/map/Storage.java]: ???

Comment (JavaPsyh) [reply src/tasks/like/map/Storage.java:3]: I don't understend you. This code everytime after "Ctrl+Alt+L".

File: src/tasks/stepic/pair/Pair.java (hunk 11 above)
Comment (JavaPsyh) [reply src/tasks/stepic/pair/Pair.java]: It spend my time, but it isn't importent. I would appreciate if I will not be distracted by unimportant issues.

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**